cmd/manager/main.go

```diff
@@ -147,64 +159,72 @@ func main() {
 	if clusterOperatorName != "" {
 		statusReporter, err = status.NewReporter(cfg, mgr, namespace, clusterOperatorName, os.Getenv("RELEASE_VERSION"), stopCh)
 		if err != nil {
-			exit(err)
+			logrus.Error(err)
+			os.Exit(1)
 		}
 	}
 
 	// Populate the global default OperatorSources definition and config
 	err = defaults.PopulateGlobals()
 	if err != nil {
-		exit(err)
+		logrus.Error(err)
+		os.Exit(1)
 	}
 
 	// Setup all Controllers
 	if err := controller.AddToManager(mgr, options.ControllerOptions{}); err != nil {
-		exit(err)
+		logrus.Error(err)
+		os.Exit(1)
 	}
 
-	// Serve a health check.
-	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusOK)
-	})
-	go http.ListenAndServe(":8080", nil)
+	logrus.Info("Setting up health checks")
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		logrus.Error(err, "unable to set up health check")
+		os.Exit(1)
+	}
+	if err := mgr.AddReadyzCheck("healthz", healthz.Ping); err != nil {
+		logrus.Error(err, "unable to set up ready check")
+		os.Exit(1)
+	}
```

Comment:
Updated the liveness and readiness probes to use the controller-runtime healthz package instead of handling it ourselves.

Reply:
I need to verify this locally as well, but something I'm realizing now is whether we'll need to bump the initial delay on the health check probes to accommodate the healthz.Ping check.